sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -133,6 +133,10 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
   private val notIncludedMsg = "[not included in comparison]"
   private val clsName = this.getClass.getCanonicalName
 
+  override def sparkConf: SparkConf = super.sparkConf
+    // Reduce shuffle partitions to reduce testing time.
+    .set(SQLConf.SHUFFLE_PARTITIONS, 5)
```

Comment:
For Python UDF test, this seems to increase from 4 to 5. Did I understand correctly?

Reply:
Actually why don't we try 4 @wangyum?

Reply:
+1 for `4`.

Reply:
OK. I'll try to set it to 4. This is because it is set to 5 in two places:
https://github.com/apache/spark/blob/359375eff74630c9f0ea5a90ab7d45bf1b281ed0/sql/core/src/test/scala/org/apache/spark/sql/test/TestSQLContext.scala#L58-L64
https://github.com/apache/spark/blob/42b80ae128ab1aa8a87c1376fe88e2cde52e6e4f/sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala#L613-L620

Reply:
`=4 ` has another issue: `--SET spark.sql.autoBroadcastJoinThreshold=10485760` will success, but `--SET spark.sql.autoBroadcastJoinThreshold=-1,spark.sql.join.preferSortMergeJoin=true` will failed:
```
22:31:31.233 ERROR org.apache.spark.sql.SQLQueryTestSuite: Error using configs: spark.sql.autoBroadcastJoinThreshold=-1,spark.sql.join.preferSortMergeJoin=true,spark.sql.codegen.wholeStage=true,spark.sql.codegen.factoryMode=CODEGEN_ONLY
[info] - subquery/in-subquery/not-in-joins.sql *** FAILED *** (32 seconds, 609 milliseconds)
[info]   subquery/in-subquery/not-in-joins.sql
[info]   Expected "1    16      12      [21
[info]   1      16      12      10]
[info]   1      10      NULL    12
[info]   1      6       8       ...", but got "1        16      12      [10
[info]   1      16      12      21]
[info]   1      10      NULL    12
[info]   1      6       8       ..." Result did not match for query #6
[info]   SELECT Count(DISTINCT( t1a )),
[info]          t1b,
[info]          t1c,
[info]          t1d
[info]   FROM   t1
[info]   WHERE  t1a NOT IN (SELECT t2a
[info]                      FROM   t2
[info]                      JOIN t1
[info]                      WHERE  t2b <> t1b)
[info]   GROUP  BY t1b,
[info]             t1c,
[info]             t1d
[info]   HAVING t1d NOT IN (SELECT t2d
[info]                      FROM   t2
[info]                      WHERE  t1d = t2d)
[info]   ORDER BY t1b DESC (SQLQueryTestSuite.scala:383)

```

Reply:
Can we add an ORDER BY to make the query output deterministic?